src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,52 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("d.MM.yyyy");
+    private static int endSum = 0;
+    private StringBuilder result = new StringBuilder();
+    private String endResult = "";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        if (!endResult.equals("")) {
+            endResult = "";
+        }
+        if (!result.isEmpty()) {
+            result.setLength(0);
+        }
+        String first = "Report for period " + dateFrom + " - " + dateTo
+                + System.lineSeparator();
+        result.append(first);
+        LocalDate beginningDate = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate endingDate = LocalDate.parse(dateTo, FORMATTER);
+        int[] numbers = new int[names.length];
+        for (int y = 0; y < names.length; y++) {
```

Comment:
hint: you can create a variable that will store the salary for specific person here:
`int salary = 0;`
Then, while iterating through data, add found values to this variable and then add entry to report.